fpdf/text_region.py

```diff
@@ -106,6 +136,30 @@ def write(self, text: str, link=None):
                 frag.link = link
         self._text_fragments.extend(fragments)
 
+    def generate_bullet_frag_and_tl(
+        self, bullet_string: str, rel_x_displacement: float
+    ):
+        if not bullet_string:
+            return None
+        if not self.pdf.font_family:
+            raise FPDFException("No font set, you need to call set_font() beforehand")
+        (bullet_fragment,) = self.pdf._preload_font_styles(
+            bullet_string, markdown=False
+        )
+        bullet_line_break = MultiLineBreak(
```

Comment:
I do not expect so (outside of very long `<ol>` bullets, which my code doesn't seem to handle anyway).
I take it, I should just opt for creating a `TextLine` object directly?

Reply:
Your current version is the most defensive way of handling it. It may look excessive, but with most bullet strings being at best a few characters long, it shouldn't incur too much of a performance penalty.
You'll just have to take into account that there can be more than one line when moving the block text out of the way (and test for that case as well).

Reply:
> You'll just have to take into account that there can be more than one line when moving the block text out of the way (and test for that case as well)

Not sure what you mean by 'block text'.

Also, it seems to me that just creating a `TextLine` directly would complicate the code, and we would still need to, essentially, deal with a `CurrentLine` the way `MultiLineBreak.get_line()` does anyway. I think I better leave this in, unless I'm missing something.

Reply:
"block text" means the text of the actual paragraph here.
Yes, you can leave in the wrapping call if that simplifies things. Just keep in mind that you may end up with wrapped lines in places where you didn't expect them.

Reply:
In the case of handling left-aligned bullets, it seems that wrapping long bullets will be needed anyway.